src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1312,6 +1313,12 @@ internal bool HasCallerInfoStringConversion(TypeSymbol destination, ref HashSet<
 
         public static bool HasIdentityConversion(TypeSymbol type1, TypeSymbol type2)
         {
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            return HasIdentityConversion(type1, type2, ref useSiteDiagnostics);
```

Comment:
HasIdentityConversion is called extremely often in the compiler with "true" result. It is the first conversion that we check for since it is the most common.
We are deferring to a much heavier method that would call Equals two times in order to possibly create a HashSet with diagnostics that we will drop here anyways.

Can we just  'return type1.Equals(type2, TypeCompareKind.AllIgnoreOptions);'  here?

If we want the one helper to call another, perhaps the heavy one should call the simpler one?